lib/actions/fcs.py

```diff
@@ -339,6 +366,422 @@ def fcs2html(self, req):
+
+        data = [
+            ('version', parameters['version']),
+            ('record', [
+                ('recordSchema', 'http://explain.z3950.org/dtd/2.0/'),
+                ('recordXMLEscaping', parameters['record_xml_escaping']),
+                ('recordData', [
+                    self._get_element_tree('zr', 'explain', [
+                        ('serverInfo', [
+                            ('host', parameters['server_name']),
+                            ('port', parameters['server_port']),
+                            ('database', parameters['database'])]),
+                        ('databaseInfo', [
+                            ('title', parameters['corpname'], {'lang': 'en', 'primary': 'true'}),
+                            ('description', default_corpus_info['corpus_desc'], {'lang': 'en'})]),
+                        ('indexInfo',
+                         [('set', [
+                             ('title', 'CLARIN Content Search', {'lang': 'en', 'primary': 'true'})
+                         ], {'identifier': 'http://clarin.eu/fcs/resource', 'name': 'fcs'})] +
+                         [('index', [('title', search_index)], {'search': 'true', 'scan': 'true', 'sort': 'true'}) for
+                          search_index in default_corpus_info['result']]),
+                        ('schemaInfo', [
+                            ('schema', [
+                                ('title', 'CLARIN Content Search', {'lang': 'en', 'name': 'fcs'})
+                            ], {'identifier': 'http://clarin.eu/fcs/resource', 'name': 'fcs'}),
+                        ]),
+                        ('configInfo', [
+                            ('default', '250', {'type': 'numberOfRecords'}),
+                            ('setting', '250', {'type': 'maximumRecords'}),
+                        ])
+                    ], {'xmlns:zr': "http://explain.z3950.org/dtd/2.0/"})
+                ]),
+                ('echoedExplainRequest', [
+                    ('version', parameters['version'])
+                ])
+            ])]
+
+        if explain_response:
+            data.append(
+                ('extraResponseData',
+                 [self._get_endpoint_description_tree(parameters)])
+            )
+
+        tree = self._get_element_tree('sruResponse', 'explainResponse', data,
+                               {'xmlns:sruResponse': "http://docs.oasis-open.org/ns/search-ws/sruResponse"})
+
+        return tree
+
+    def _get_default_corpora(self):
+        # Returns kontext default corpora from settings
+
+        default_corp_list = settings.get('corpora', 'default_corpora', [])
+        corpname = None
+        if 0 == len(default_corp_list):
+            _logger.critical('FCS cannot work properly without a default_corpora set')
+        else:
+            corpname = default_corp_list[0]
+        return corpname
+
+    def _get_search_retrieve_tree(self, parameters, results, number_of_records, start_row=1, max_records=250, ns='sru'):
```

Comment:
`start_row`, `max_record`, `ns` seem unused

Reply:
I missed `start_row` and `max_record`, `ns` is deleted